Makefile

```diff
@@ -38,7 +38,7 @@ format-check:
 	isort --check-only $(ISORT_CONFIG) $(ISORT_TARGETS)
 
 type-check:
-	mypy .
+	mypy gpflow tests
```

Comment:
there's also a mention of mypy in the contrib guidelines.
> Use `pylint` and `mypy` for formatting and _type checking_
I don't know if you want to clarify there that the notebooks don't need to pass. Whatever you think makes more sense